e2e/cypress/integration/channel/message_permalink.js

```diff
@@ -0,0 +1,105 @@
+import users from '../../fixtures/users.json';
+
+const cypressConfig = require('../../../cypress.json');
+
+describe('Message permalink', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M13675-Copy a permalink and paste into another channel', () => {
+        const message = 'Hello' + Date.now();
+        const channelName = 'test-message-channel-1';
+
+        // # Create new DM channel with user's email
+        cy.apiGetUsers(['user-1', 'sysadmin']).then((userResponse) => {
+            const userEmailArray = [userResponse.body[1].id, userResponse.body[0].id];
+
+            cy.apiCreateDirectChannel(userEmailArray).then((response) => {
+                const direcrChannel = response.body;
+                cy.getCurrentTeamName().then((teamName) => {
+                    var currentTeamName = `${teamName}`.toLowerCase();
+                    const url = `${cypressConfig.baseUrl}/${currentTeamName}/messages/@${users.sysadmin.username}`;
+                    cy.visit(url);
+                    cy.get(`#sidebarItem_${direcrChannel.name}`).click({force: true});
+
+                    // # Post message to use
+                    cy.postMessage(message + '{enter}');
+
+                    cy.getLastPostId().then((postId) => {
+                        // # check if ... button is visible in last post right side
+                        cy.get(`#CENTER_button_${postId}`).should('not.be.visible');
+
+                        // # click on ... button of last post
+                        cy.clickPostDotMenu(postId);
+
+                        // spy on function and verify it
+                        cy.document().then((doc) => {
+                            cy.spy(doc, 'execCommand').as('execCommandSpy');
+                        });
+
+                        // # click on permalink option
+                        cy.get(`#permalink_${postId}`).should('be.visible').click();
+
+                        // # copy permalink from button
+                        cy.get('#linkModalCopyLink').click();
+
+                        // # Copy link url from text area
+                        cy.get('#linkModalTextArea').invoke('val').as('linkText');
+
+                        cy.get('@execCommandSpy').should('have.been.calledWith', 'copy');
+
+                        // # close modal dialog
+                        cy.get('#linkModalCloseButton').click();
+                    });
+                });
+            });
+        });
+
+        // # get current team id
+        cy.getCurrentTeamId().then((teamId) => {
+            // # create public channel to post permalink
+            cy.apiCreateChannel(teamId, channelName, channelName, 'O', 'Test channel').then((response) => {
+                const testChannel = response.body;
+
+                // # click on test public channel
+                cy.get('#sidebarItem_' + testChannel.name).click();
```

Comment:
Please add option to force click to prevent flaky test whenever there's too many newly created channels that makes this out of the view.
```suggestion
                cy.get('#sidebarItem_' + testChannel.name).click({force: true});
```